### DISCUSSION TITLE
OsgEarth cache

### DISCUSSION THREAD

**Discussion by JD31** [pelicanmapping/osgearth#1875] [Support Q&A]
Hi,

With OsgEarth 2.10.2 when using cache with a 'line' type feature model (which is rendered as LineDrawable) I observe that the number of stateset / vertices / ... is increased.
![image](https://user-images.githubusercontent.com/17981960/142823808-31782950-eae7-4e3c-97c3-45ad706ddeb4.png)
Like shown in the image above, it is like if everything is duplicated. It seems to be an instancing issue.
Any ideas of what is causing that and how to improve?

Regards,
JD.

**Comment by gwaldron** [comment]
Does osgEarth 3.2 exhibit this same behavior?
Will you provide a simple earth file for us to reproduce?

**Reply by JD31** [reply]
Unfortunatly I can't test easily 3.2.
Please find attached a zip file with the data and the earth file for testing.

[lineCache.zip](https://github.com/gwaldron/osgearth/files/7641261/lineCache.zip)

**Comment by JD31** [comment]
I found the reason of the issue.
In the serializer code of LineGroup, osg::Group shall be removed from the serializer chain.
`"osg::Object osg::Node osg::Group osg::Geode osgEarth::LineGroup"`
to
`"osg::Object osg::Node osg::Geode osgEarth::LineGroup"`

I guess there is a kind of redundancy in the code  for Geode and Group with regard to their children management.
If looking to osg geode serializer we can see that there is no ancestor call to osg::Group serializer.

The bug is not applicable to current master code as the LineGroup is now only a Group, instead of a Geode.
